src/env.h

```diff
@@ -1101,6 +1116,7 @@ class Environment final : public MemoryRetainer {
   TickInfo tick_info_;
   permission::Permission permission_;
   const uint64_t timer_base_;
+  Cleanable* cleanables_;
```

Comment:
The bare pointer makes me a bit nervous. This should have some guard rails around it to prevent UAFs